openapi/paths/api@courses.yaml

```diff
@@ -0,0 +1,29 @@
+get:
+  operationId: searchCourses
+  tags: [course]
+  summary: Get a list of courses
+  description: |
+    This allows you to search for courses get details about the course. The search filter attributes are all optional.
+    If the limit query parameter is not provided, the number of courses returned is defaulted to 100.
+    This endpoint is only available to superusers and staff
+
+  responses:
+    '200':
+      $ref: ../components/responses/course-list.yaml
+    default:
+      $ref: ../components/responses/problem.yaml
+  parameters:
+    - name: query
+      in: query
+      description: |
+        Will do a fuzzy match on the `course` name
+      schema:
+        type: string
+    - name: folder
+      in: query
+      description: |
+        Passing folder ID will return related courses as suggestions
+      schema:
+        type: integer
+    - $ref: ../components/parameters/cursor.yaml
+    - $ref: ../components/parameters/limit.yaml
```

Comment:
depending on what we decide, we may need a note on the limit / cursor is only for the all courses array, and not the suggestions array.